### PR TITLE
Sound setting improvements

### PR DESCRIPTION
**Pull request by jquanton**
## Description
Changed sound effects to only play for:
- When user intervention is required
- On errors
- On task completion

Added volume control for sound effects. Users can now adjust the volume level of notification sounds through a slider in settings when sound effects are enabled.
Screenshot 2024-12-16 at 12 40 46 AM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] My code follows the patterns of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Reviewers

----

> [!IMPORTANT]
> Improves sound settings by adding volume control and refining sound triggers, with corresponding tests.
> 
>   - **Behavior**:
>     - Sound effects now play only for user intervention, errors, and task completion in `ChatView.tsx`.
>     - Added volume control for sound effects in `SettingsView.tsx`.
>   - **Settings**:
>     - Added `soundVolume` state and control in `ExtensionStateContext.tsx` and `SettingsView.tsx`.
>     - Updated `setSoundVolume()` in `sound.ts` to adjust sound volume.
>   - **Testing**:
>     - Added tests for sound playing behavior in `ChatView.test.tsx`.
>     - Added tests for sound settings in `SettingsView.test.tsx`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooVetGit%2FRoo-Cline&utm_source=github&utm_medium=referral)<sup> for ca4806db224ba1b702deced67ee01139cbb8d18e. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by mrubens** [comment]
Nice! Just trying it out now. Seems like it's still beeping on auto-approved execute commands.

File: package.json [hunk 1]
```diff
@@ -220,6 +220,7 @@
     "puppeteer-chromium-resolver": "^23.0.0",
     "puppeteer-core": "^23.4.0",
     "serialize-error": "^11.0.3",
+    "sound-play": "^1.1.0",
```
Comment (mrubens) [inline package.json:223]: Can we remove play-sound?

**Comment by mrubens** [comment]
I wonder if we should simplify and just beep on questions, errors, and completion, and not need to bother with all of the autoapprove logic - it feels kind of brittle and I'm not sure people want to hear sounds when they're directly supervising Cline anyway?

File: package.json (hunk 1 above)
Comment (jquanton) [reply package.json:222]: Done ✅

**Comment by jquanton** [comment]
> Nice! Just trying it out now. Seems like it's still beeping on auto-approved execute commands and MCP use.

Pushed a few changes, can you try it out now?

**Review by mrubens — APPROVED** [review]
Code looks great, and it seems to work well! Only tiny bug is that it does a double-beep when prompting you to use a MCP tool that isn't auto-approved. Is that an easy fix?

**Comment by jquanton** [comment]
> Code looks great, and it seems to work well! Only tiny bug is that it does a double-beep when prompting you to use a MCP tool that isn't auto-approved. Is that an easy fix?

Pushed a fix

**Review by mrubens — APPROVED** [review]

**Pull request merged**